### PR TITLE
add "zar zq"

### PR DESCRIPTION
**Pull request by mccanne**
This command adds support for the "zq" sub-command of zar, which
lets you run zq-style commands over the log files in a zar archive.
In this case, the file "_" is the log file in question and any
files referred to live within the zar directory associated with
the log.  The zq command is run for each log file found.

### CODE REVIEW

**Comment by mccanne** [comment]
I will add tests in the morning...

File: cmd/zar/zq/command.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+			}
+			inputs = inputs[1:]
+		}
+		localPaths := archive.Localize(zardir, inputs)
+		readers, err := c.inputReaders(resolver.NewContext(), localPaths)
+		if err != nil {
+			return err
+		}
+		if len(readers) == 0 {
+			// XXX silently skip if no file found
```
Comment (alfred-landrum) [inline cmd/zar/zq/command.go:161]: Skipping & proceeding seems like the right thing to do; do you think we should log here?

File: cmd/zar/zq/command.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+			}
+			err = fmt.Errorf("%s: %w", path, err)
+			if c.stopErr {
+				return nil, err
+			}
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+			continue
+		}
+		// wrap in a named reader so the reader implements Stringer
+		readers = append(readers, namedReader{file, path})
```
Comment (alfred-landrum) [inline cmd/zar/zq/command.go:213]: I don't see the String() method used, is that for debugging?

File: cmd/zar/zq/command.go [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/brimsec/zq/zql"
+	"github.com/mccanne/charm"
+	"go.uber.org/zap"
+)
+
+var Zq = &charm.Spec{
+	Name:  "zq",
+	Usage: "zq [-R dir] [options]",
+	Short: "walk an archive and run zq logic",
```
Comment (henridf) [inline cmd/zar/zq/command.go:34]: what does "logic" mean here ... could this be more simply expressed as "walk an archive and run zq over it"? (similar comment for the `Long:` section below)

File: cmd/zar/zq/command.go [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+		Format:         c.ReaderFlags.Format,
+		DashStdin:      true,
+		JSONTypeConfig: c.jsonTypeConfig,
+		JSONPathRegex:  c.jsonPathRegexp,
+	}
+	var readers []zbuf.Reader
+	for _, path := range paths {
+		file, err := detector.OpenFile(zctx, path, cfg)
+		if err != nil {
+			// XXX for zar zq siliently conotinue if file doesn't exist
```
Comment (henridf) [inline cmd/zar/zq/command.go:201]: siliently conotinue

File: cmd/zar/zq/command.go [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+	// Flags added for writers are -f, -T, -F, -E, -U, and -b
+	c.WriterFlags.SetFlags(f)
+
+	// Flags added for readers are -i XXX json
+	c.ReaderFlags.SetFlags(f)
+
+	return c, nil
+}
+
+func (c *Command) loadJsonTypes() (*ndjsonio.TypeConfig, error) {
```
Comment (henridf) [inline cmd/zar/zq/command.go:96]: can we get rid of all this json stuff here? or do that as part of the "xxx refactor into a tools package" comment?

File: cmd/zar/zq/command.go (hunk 1 above)
Comment (mccanne) [reply cmd/zar/zq/command.go:161]: changing to warn to stderr in no -q

File: cmd/zar/zq/command.go (hunk 2 above)
Comment (mccanne) [reply cmd/zar/zq/command.go:213]: scanner.Combiner.Read() calls *printf with the reader interface pointer as a %s.   I think Noah came up with this.

Comment (mccanne) [reply cmd/zar/zq/command.go:213]: I will clarify the comment

File: cmd/zar/zq/command.go (hunk 3 above)
Comment (mccanne) [reply cmd/zar/zq/command.go:34]: I'll tweak it.  It doesn't "run zq".  It runs zql queries similar to how the standalone zql command works.

File: cmd/zar/zq/command.go (hunk 4 above)
Comment (mccanne) [reply cmd/zar/zq/command.go:201]: will put a warnring for no -q

File: cmd/zar/zq/command.go (hunk 5 above)
Comment (mccanne) [reply cmd/zar/zq/command.go:96]: I would rather leave it as a placeholder so we remember to adjust this call site.  The log files here referred to with _ can be json in need of schema maps.

**Pull request merged**